[examples, book] adds ID Pointer pattern example (#5248)
MystenLabs/sui: doc/book/src/patterns/id-pointer.md

# ID Pointer

> ID Pointer is a technique which allows separation of the main data (an object) and its accessors / capabilities by linking the latter to the original. There's a few different directions in which this pattern can be used:

Reviewer: ```suggestion
ID Pointer is a technique that separates the main data (an object) and its accessors / capabilities by linking the latter to the original. There's a few different directions in which this pattern can be used:
```

Is there a different word for accessors? And it is odd to say that it separates the data by linking the things together. Is isolate correct? Is it that ID Pointer creates a link between objects and accessors?

- issuing transferable capabilities for shared objects (for example, a TransferCap that changes 'owner' field of a shared object)

> - splitting dynamic data and static (eg NFT and its Collection information)

Reviewer: dynamic data and static data?
```suggestion
- splitting dynamic data and static (for example, an NFT and its Collection information)
```
why is Collection upper case? The way this is worded the NFT is the dynamic data?

Author: Uppercase because it's a Type.

- avoiding unnecessary type linking (and witness requirement) in generic applications (LP token for a LiquidityPool)

```move
{{#include ../../examples/sources/patterns/id_pointer.move:4:}}
```

> This pattern is used in these examples:

Reviewer: ```suggestion
The following examples demonstrate using this pattern:
```

Author: I'll keep it as is to align with everything else (on every other page we have this wording).
